test/controllers/contacts_controller_test.rb

```diff
@@ -0,0 +1,18 @@
+require "test_helper"
+
+class ContactsControllerTest < ActionDispatch::IntegrationTest
+  test "should get new" do
+    get contacts_new_url
+    assert_response :success
+  end
+
+  test "should get show" do
+    get contacts_show_url
+    assert_response :success
+  end
+
+  test "should get index" do
+    get contacts_index_url
+    assert_response :success
+  end
```

Comment:
What is the difference between this and the contacts spec.rb testing the routes?

Reply:
But will delete